specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json

```diff
@@ -1762,16 +1824,68 @@
           "$ref": "#/definitions/ManagedClusterAADProfile",
           "description": "Profile of Azure Active Directory configuration."
         },
-        "apiServerAuthorizedIPRanges": {
+        "privateLinkResources": {
           "type": "array",
           "items": {
-            "type": "string"
+            "$ref": "#/definitions/ManagedClusterPrivateLinkResource"
           },
-          "description": "(PREVIEW) Authorized IP Ranges to kubernetes API server."
+          "description": "Properties related to private link cluster."
+        },
+        "apiServerAccessProfile": {
+          "$ref": "#/definitions/ManagedClusterAPIServerAccessProfile",
+          "description": "Access profile for managed cluster API server."
         }
       },
       "description": "Properties of the managed cluster."
     },
+    "ManagedClusterPrivateLinkResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The fully qualified Azure resource ID."
+        },
+        "name": {
+          "type": "string",
+          "description": "The private link resource name."
+        },
+        "type": {
+          "type": "string",
+          "description": "The private link resource type."
+        },
+        "groupId": {
```

Comment:
schema of private link resource must comply with that of ARM resource. id, name, type top level properties and then all other properties must be under "properties" property. Is the privateLinkResource being created as part of creation of managedCluster? I am assuming not. And so, if this is just a reference to an existing privateLinkResource, why is the reference not done via just the resourceid. Is there an RBAC element to this as well? The way you have modeled it, someone who does not have read access to the private endpoint resource will be able to read all information about it. 
Apologies for my limited understanding of how private link resource in NRP works.  I am assuming there must be a standard contract between NRP and every RP that wants to leverage private links.

Reply:
@weinong  your input is needed here

Reply:
we decided to remove this in this version.